pkg/providers/amifamily/bootstrap/nodeadm.go

```diff
@@ -115,11 +116,11 @@ func (n Nodeadm) generateInlineKubeletConfiguration() (map[string]runtime.RawExt
 	if err != nil {
 		return nil, err
 	}
-	if len(n.Taints) != 0 {
-		kubeConfigMap["registerWithTaints"] = runtime.RawExtension{
-			Raw: lo.Must(json.Marshal(n.Taints)),
-		}
+	n.Taints = append(n.Taints, core.Taint{Key: "karpenter.sh/unregistered", Value: "true", Effect: core.TaintEffectNoExecute})
```

Comment:
Will update to use taint defined in `kubernetes-sigs/karpenter` once it is merged.